did-pkh-method-draft.md

```diff
@@ -0,0 +1,256 @@
+  "publicKeyJwk": {
+    "@id": "https://w3id.org/security#publicKeyJwk",
+    "@type": "@json"
+  },
+  "Ed25519VerificationKey2018": "https://w3id.org/security#Ed25519VerificationKey2018",
+  "Ed25519PublicKeyBLAKE2BDigestSize20Base58CheckEncoded2021": "https://w3id.org/security#Ed25519PublicKeyBLAKE2BDigestSize20Base58CheckEncoded2021",
+  "P256PublicKeyBLAKE2BDigestSize20Base58CheckEncoded2021": "https://w3id.org/security#P256PublicKeyBLAKE2BDigestSize20Base58CheckEncoded2021",
+  "TezosMethod2021": "https://w3id.org/security#TezosMethod2021",
+  "EcdsaSecp256k1RecoveryMethod2020": "https://identity.foundation/EcdsaSecp256k1RecoverySignature2020#EcdsaSecp256k1RecoveryMethod2020"
+}
+```
+## Operations
+
+### Create
+
+The blockchain account id is validated according to [CAIP-10][] and then appended to
+`did:pkh:{network}:`, where `{network}` is the supported prefix corresponding to
+the blockchain where it is valid.
+
+### Read (Resolve)
+
+Resolution implements the following interface defined in [DID Core][]:
+```
+resolve(did, resolutionOptions) →
+   « didResolutionMetadata, didDocument, didDocumentMetadata »
+```
+
+Construct the DID Document for *did* as follows:
+- Parse the DID into its network id, *network* and account address, *address*,
+  according to Syntax and Interpretation above.
+- Initialize a DID document, *doc*, as a JSON-LD document.
+- Set the `id` property of *doc* to *did*.
+- Set the `@context` property of *doc* to an array,
+  `["https://www.w3.org/ns/did/v1", context]`, where `context` is the [did:pkh JSON-LD context object](#context).
+- Construct the verification method ID, *vm*, by appending
+  "#blockchainAccountId" to *did*.
+- Construct the [verification method][] object, *vmObj* as follows:
+  - Insert property `id` into *vmObj* with value *vm*.
+  - Look up network id *network* in the did:pkh [Networks][#networks] table, to
+    get verification method type *vmType* and [CAIP-2][] chain id *chainId*. If
+    there are multiple entries in the table for *network*, use one that
+    matches *address*.
+  - Insert property `type` into *vmObj* with value *vmType*.
+  - Insert property `controller` into *vmObj* with value *did*.
+  - Construct string *accountId* by concatenating *address* + "@" + *chainId*.
+  - Insert property `blockchainAccountId` into *vmObj* with value *accountId*.
+- Insert a property into *doc* with key name `verificationMethod` and a value of
+  an array containing only *vmObj*.
+- Insert a property into *doc* with key name `authentication` and a value of an
+  array containing only *vm*.
+- Insert a property into *doc* with key name `assertionMethod` and value of an
+  array containing only *vm*.
+- Construct an empty DID Resolution metadata object, *resMeta*.
+- Construct an empty DID Document metadata object, *docMeta*.
+- Return *resMeta*, *doc*, and *docMeta*.
+
+
+### Update
+
+No updates possible. did:pkh DID Documents are, like [did:key] documents, intended for local-only usage.
```

Comment:
What does "local-only usage" mean here?

Reply:
the did:key spec says to only use it "local-only", i.e., if you don't have the public key on hand/in-memory, don't trust it, and don't send the derived DID document far into the world.  I used the did:key spec as a template, and inherited this term uncritically. what would make more sense for did:pkh? "intended only for trust-models where a PKH is separately authenticated and revoked from the DID document derived from it"?

Reply:
For trust models where trust is inherited from the PKH being observed on-chain?